client.go

```diff
@@ -208,6 +212,8 @@ type Client struct {
 	contentTypeDecoders      map[string]ContentTypeDecoder
 	contentDecompressorKeys  []string
 	contentDecompressors     map[string]ContentDecompressor
+	stopChan                 chan bool
+	certLock                 *sync.Mutex
```

Comment:
@cdoucy In Resty v3, RWMutex was added at the client level to protect all read/write operations on client fields. Please use that lock instead of creating a new one for the cert alone.

Reply:
I removed it.

Since `handleCAs` is now goroutine safe, I don't even to acquire the lock in the cert watcher loop :)